dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/CategoryComboModificationControllerTest.java

```diff
@@ -0,0 +1,216 @@
+                + "'period':'20220102',"
+                + "'orgUnit':'%s',"
+                + "'value':'24',"
+                + "'comment':'OK'}",
+            dataElementId, categoryOptionComboId, orgUnitId);
+
+    assertStatus(HttpStatus.CREATED, POST("/dataValues", body));
+
+    // We should not be able to remove a category here
+    assertStatus(
+        HttpStatus.CONFLICT,
+        PUT(
+            "/categoryCombos/" + testCatCombo + "?mergeMode=REPLACE",
+            "{ 'name' : 'COLOR AND TASTE', 'id' : '"
+                + testCatCombo
+                + "', "
+                + "'shortName': 'C_AND_T', 'skipTotals' : true, "
+                + "'dataDimensionType' : 'DISAGGREGATION', 'categories' : ["
+                + "{'id' : '"
+                + categoryColor
+                + "'} ]} "));
+  }
+
+  void setTestCatComboModifiableProperties() {
+    // Modify the initial name
+    assertStatus(
+        HttpStatus.OK,
+        PUT(
+            "/categoryCombos/" + testCatCombo + "?mergeMode=REPLACE",
+            "{ 'name' : 'COLOR AND TASTE', 'id' : '"
+                + testCatCombo
+                + "', "
+                + "'dataDimensionType' : 'DISAGGREGATION', 'categories' : ["
+                + "{'id' : '"
+                + categoryColor
+                + "'} , {'id' : '"
+                + categoryTaste
+                + "'}]} "));
+    // Add a shortname
+    assertStatus(
+        HttpStatus.OK,
+        PUT(
+            "/categoryCombos/" + testCatCombo + "?mergeMode=REPLACE",
+            "{ 'name' : 'COLOR AND TASTE', 'id' : '"
+                + testCatCombo
+                + "', "
+                + "'shortName': 'C_AND_T', "
+                + "'dataDimensionType' : 'DISAGGREGATION', 'categories' : ["
+                + "{'id' : '"
+                + categoryColor
+                + "'} , {'id' : '"
+                + categoryTaste
+                + "'}]} "));
+    // Skip totals
+    assertStatus(
+        HttpStatus.OK,
+        PUT(
+            "/categoryCombos/" + testCatCombo + "?mergeMode=REPLACE",
+            "{ 'name' : 'COLOR AND TASTE', 'id' : '"
+                + testCatCombo
+                + "', "
+                + "'shortName': 'C_AND_T', 'skipTotals' : true, "
+                + "'dataDimensionType' : 'DISAGGREGATION', 'categories' : ["
+                + "{'id' : '"
+                + categoryColor
+                + "'} , {'id' : '"
+                + categoryTaste
+                + "'}]} "));
+  }
+
+  void setupTest() {
+    String categoryOptionSour =
+        assertStatus(
+            HttpStatus.CREATED,
+            POST("/categoryOptions", "{ 'name': 'Sour', 'shortName': 'Sour' }"));
+
+    String categoryOptionRed =
+        assertStatus(
+            HttpStatus.CREATED, POST("/categoryOptions", "{ 'name': 'Red', 'shortName': 'Red' }"));
+
+    categoryColor =
+        assertStatus(
+            HttpStatus.CREATED,
+            POST(
```

Comment:
I think you should add a helper method for creating categories with essentially this part where you only pass in the body. This will make this much more readable as you don't get the breakdown like this. Also you can use text-blocks again for the body ;)